File: japan-go-be/src/main/java/com/japan_go_be/features/lesson/dto/request/LessonRequest.java
package com.japan_go_be.features.lesson.dto.request;

import com.japan_go_be.features.lesson.constant.LessonTypeEnum;

public record LessonRequest(
        Long lessonId,
        Long folderId,
        String lessonName,
        String description,
        LessonTypeEnum lessonType,
        GrammarLessonRequest grammarLesson,
        KanjiLessonRequest kanjiLesson
) {
}

File: japan-go-fe/types/api/requests/lesson.request.ts
import { KanjiPageRequest } from "@/types/api/requests/kanji.request";
import { LessonType } from "@/types/enums/lesson.enum";

export interface FolderLessonRequest {
    folderId: number | string;
    lessonId: number | string;
}

export interface KanjiLessonRequest {
    lessonId?: number;
    folderId: number | null;
    bookId: number;
    lessonType: LessonType;
    lessonName: string;
    description: string;
    kanjiPages: KanjiPageRequest[];
}

export interface GrammarLessonRequest {
    folderId: number | null;
    jlptLevel: number;
    bookId: number;
    files: File[];
}

export interface ExportGrammarLessonRequest {
    grammarLessonIds: number[];
}
